packages/SwingSet/src/kernel/kernel.js

```diff
@@ -308,33 +308,35 @@ export default function buildKernel(
     }
   }
 
-  function removeVatManager(vatID, reason) {
+  function removeVatManager(vatID, shouldReject, info) {
+    insistCapData(info);
     const old = ephemeral.vats.get(vatID);
     ephemeral.vats.delete(vatID);
-    const err = reason ? Error(reason) : undefined;
-    old.notifyTermination(err);
+    old.notifyTermination(shouldReject, info);
     return old.manager.shutdown();
   }
 
-  function terminateVat(vatID, reason) {
+  function terminateVat(vatID, shouldReject, info) {
+    insistCapData(info);
     if (kernelKeeper.getVatKeeper(vatID)) {
       const promisesToReject = kernelKeeper.cleanupAfterTerminatedVat(vatID);
-      const err = reason ? makeError(reason) : VAT_TERMINATION_ERROR;
+      const err = VAT_TERMINATION_ERROR;
```

Comment:
nit: let's fold this into the call, so make line 325 into `resolveToError(kpid, VAT_TERMINATION_ERROR, vatID);`, unless that makes the line so long that lint wants to fold it

Reply:
Yup.   Palimpsest code.